pkg/operator/controller/status/controller.go

```diff
@@ -472,24 +479,17 @@ func computeOperatorProgressingCondition(haveDNS bool, dns *operatorv1.DNS, oldV
 		}
 	}
 
-	upgrading := false
-	for name, curVersion := range curVersions {
-		if oldVersion, ok := oldVersions[name]; ok && oldVersion != curVersion {
-			messages = append(messages, fmt.Sprintf("Upgraded %s to %q.", name, curVersion))
-		}
-		if newVersion, ok := newVersions[name]; ok && curVersion != newVersion {
-			upgrading = true
-			messages = append(messages, fmt.Sprintf("Upgrading %s to %q.", name, newVersion))
-		}
-	}
+	// If the operator is upgrading, note it as a Progressing reason and add the upgrading messages
+	upgrading, upgradingMessages := isUpgrading(curVersions, oldVersions, newVersions)
```

Comment:
Initially, I thought this meant "Cluster is Upgrading". Seems like that isn't the case, which I think you are aware of, based on the code comment above. This looks to based on the version of:
1. Operator
2. CoreDNS Pods
3. KubeRBac roxy
4. OpenshiftCLI (I'm not sure, but I think this is an image that the NodeResolver daemon set uses)

I don't I have any issues with that, but it sounds like it's been implied that this is resolving [OCPBUGS-14346](https://issues.redhat.com//browse/OCPBUGS-14346), but I wonder if it's worth pointing out that it's only part of the problem.

AFAIK, the degrade status comes from two types of situations during upgrade:
1. DNS-related pods (the four mentioned above) are rolling out due to new versions
2. The nodes are rolling out (near the end of the upgrade)

Both situations result in DNS pods going unavailable. Unless I'm misunderstanding, this `isUpgrading` doesn't account for the node rebooting/rollout portion of the upgrade. Was the intention for `IsUpgrading` to also protect against degraded status during the node rollout? I am misunderstanding the problem?

Reply:
Afaik, the concern is whether the operator is upgrading, not whether the cluster is upgrading.